### DISCUSSION TITLE
CSV, Parquet and column based JSON output format

### DISCUSSION THREAD

**Discussion by burma69** [Azure/data-api-builder#1894] [Ideas]
As JSON is not the best bulk data exchange format, please consider:
1) streaming CSV output (start sending as soon as the database starts sending output)
2) JSON as a collection of arrays of values per column. Some visualization libraries e.g. chart.js use JSON objects with arrays of values per series (series = column). This also reduces the payload size.
3) compressed parquet

**Comment by yorek** [comment]
Apache Arrow Flight was something we are also thinking as a more performant way to send data back and forth
